### PR TITLE
feat: Release event loop in synchronous signals

### PR DESCRIPTION
**Pull request by dj-stormtrooper**
I measured timings of these sources before the changes were implemented, details are in Notion

### CODE REVIEW

File: src/sources/dom_blockers.ts
Comment (Finesse) [inline src/sources/dom_blockers.ts:359]: Looks like this may cause the elements to be displayed on the page for a moment. Probably, `root` should be hidden with CSS such way that `offsetParent` keeps working.

Comment (dj-stormtrooper) [reply src/sources/dom_blockers.ts:359]: We can skip this source to eliminate the risk, considering low profit of this optimisation

Comment (Finesse) [reply src/sources/dom_blockers.ts:359]: Sounds good

Comment (Finesse) [reply src/sources/dom_blockers.ts:359]: The other DOM sources may have the same issues. `visibility: hidden` will suffice to solve them.

Comment (dj-stormtrooper) [reply src/sources/dom_blockers.ts:359]: Ok, I'll check

**Review by Finesse — APPROVED** [review]

**Comment by dj-stormtrooper** [comment]
Please do not merge yet, I want to check stability of these signals, I suspect that it's worse than before

**Comment by dj-stormtrooper** [comment]
Yes, the stability isn't great (30 runs for each test)

image

Let's discard the idea, too many different issues for this task

**Comment by Finesse** [comment]
> Yes, the stability isn't great (30 runs for each test)

This is weird. Was it better before the changes?

**Comment by dj-stormtrooper** [comment]
> > Yes, the stability isn't great (30 runs for each test)
> 
> This is weird. Was it better before the changes?

I think you're right, before the changes the behaviour is similar. It's just harder to notice because we use retries on CI, but locally it's the same

File: src/sources/canvas.ts
Comment (Finesse) [inline src/sources/canvas.ts:10]: Since the source is now asynchronous, please move it to the list of asynchronous sources in `src/sources/index.ts`. Same in Pro agent.

**Review by Finesse — CHANGES_REQUESTED** [review]

File: src/sources/canvas.ts
Comment (dj-stormtrooper) [reply src/sources/canvas.ts:10]: Fixed (I'll update the Pro once we merge this PR)

**Review by Finesse — APPROVED** [review]

**Pull request merged**
